tests/background_workload_test.py

```diff
@@ -0,0 +1,176 @@
+# Copyright 2015 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for background workload"""
+
+import unittest
+import os
+import mock
+from mock import patch
+from tests import mock_flags
+from perfkitbenchmarker import configs
+from perfkitbenchmarker.linux_benchmarks import ping_benchmark
+from perfkitbenchmarker import benchmark_spec
+from perfkitbenchmarker import timing_util
+from perfkitbenchmarker import pkb
+
+
+NAME = 'ping'
+UID = 'name0'
+
+CONFIG_WITH_BACKGROUND_CPU = """
+ping:
+    description: Benchmarks ping latency over internal IP addresses
+    vm_groups:
+      vm_1:
+        vm_spec:
+          GCP:
+            machine_type: n1-standard-1
+      vm_2:
+        vm_spec:
+          GCP:
+            background_cpu_threads: 3
+            machine_type: n1-standard-1
+"""
+
+
+class TestBackgroundWorkload(unittest.TestCase):
+
+  def testBackgroundWorkloadInPKB(self):
+    """ Check that benchmark spec methods are called by pkb"""
+    with mock_flags.PatchFlags() as mocked_flags:
+
+      mocked_flags.os_type = 'debian'
+      mocked_flags.machine_type = None
+      mocked_flags.cloud = 'GCP'
+      mocked_flags.background_cpu_threads = 1
+
+      collector = mock.MagicMock()
+      timer = timing_util.IntervalTimer()
+      spec = mock.MagicMock(spec=benchmark_spec.BenchmarkSpec)
+      vm0 = mock.MagicMock()
+      vm1 = mock.MagicMock()
+      spec.vms = [vm0, vm1]
+      path = os.path.join(os.path.dirname(__file__), 'data', 'ping.out')
+      outfile = open(path, 'r')
+      pingstdout = outfile.read()
+      for vm in spec.vms:
+        vm.RemoteCommand.side_effect = [(pingstdout, ' ')]
+
+      pkb.DoPreparePhase(ping_benchmark, NAME, spec, timer)
+      self.assertEquals(spec.Prepare.call_count, 1)
+
+      pkb.DoRunPhase(ping_benchmark, NAME, spec, collector, timer)
+      self.assertEquals(spec.StopBackgroundWorkload.call_count, 1)
+      self.assertEquals(spec.StartBackgroundWorkload.call_count, 1)
+
+  def testBackgroundWorkloadSpec(self):
```

Comment:
This method seems similar to the one before it. Could you combine them and mock the benchmark methods in a way that they still run? For example, using side_effect to specify the original function.